ddtrace/tracer/textmap.go

```diff
@@ -542,7 +536,17 @@ func (*propagatorB3SingleHeader) injectTextMap(spanCtx ddtrace.SpanContext, writ
 		return ErrInvalidSpanContext
 	}
 	sb := strings.Builder{}
-	sb.WriteString(fmt.Sprintf("%016x-%016x", ctx.traceID, ctx.spanID))
+	var traceID string
+	if strings.Trim(ctx.traceID128, "0") == "" { // 64-bit trace id
+		traceID = fmt.Sprintf("%016x", ctx.traceID)
+	} else { // 128-bit trace id
+		var w3Cctx ddtrace.SpanContextW3C
```

Comment:
Do the w3 names and types here make sense in the B3 propagator?

Reply:
The type `ddtrace.SpanContextW3C` is basically the name I gave it to just denote that it's a W3C compliant span. We could certainly change it, but that's why I'm naming the variable `w3Cctx` here, because it's just mapping to the type. Also happy to rename it in a separate PR if you can come up with a better name thatn `SpanContextW3C` (it's also something that will not exist in the V2 API, as we'll just merge it into our regular span type most likely)